### PR TITLE
Added support for multiple / variable backup times.

### LINKED ISSUE #91: Scheduling the time the backup is ran
- Module version: 1.6.0

## How to reproduce (e.g Puppet code you use)
`
class{'borg':
    backupserver => $targetserver,
    includes     => [
                      '/home',
                    ],
    excludes     => [
                      '/',
                      '/boot',
                      '/boot/efi',
                      '/tmp',
                      '/sys',
                      '/dev',
                      '/proc',
                      '/run',
                      '/media',
                      '/var/lib/nfs/rpc_pipefs',
                    ],
  }
`
## What are you seeing
The backup is created at 18:30 every day.

## What behaviour did you expect instead
I want the time to be configurable while calling the class, at the moment, it's hardcoded in the service definition.

### PR DESCRIPTION
**Pull request by sebastianberm**
#### Pull Request (PR) description
This PR will allow users to set the times where systemd would need to run the backup, this will also allow multiple backups to be run each day, so the load can be distributed.
Furthermore, the original backup time is no longer hardcoded at 18:30:00.

#### This Pull Request (PR) fixes the following issues
Fixes #91

### CODE REVIEW

File: REFERENCE.md [hunk 1]
```diff
@@ -257,3 +257,11 @@ configure your most favourite ssh key type. This will be used to connect to the
 
 Default value: `'ed25519'`
 
+##### `backuptime`
+
+Data type: `Hash[String,String]`
```
Comment (bastelfreak) [inline REFERENCE.md:262]: this file is autogenerated from the comments in init.pp. Please update only the init.pp and run `bundle exec rake reference` afterwards.

File: manifests/init.pp [hunk 2]
```diff
@@ -123,6 +126,7 @@
   Optional[String[1]] $proxy_server                        = undef,
   Boolean $manage_package                                  = true,
   Enum['rsa', 'ed25519'] $ssh_key_type                     = 'ed25519',
+  Hash[String,String] $backuptime                          = { 'default' => '18:30:00' },
```
Comment (bastelfreak) [inline manifests/init.pp:129]: We can also update this a bit I guess:
```suggestion
  Hash[String[1],String[1]] $backuptime                          = { 'default' => '18:30:00' },
```

This won't prohibit broken configs, but at least prevents empty strings

File: templates/borg-backup.timer.epp [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
 [Unit]
-Description=Daily Timer for borg-backup
+Description=Daily Timer(s) for borg-backup
 
 [Timer]
 # OnBootSec=10min
 # OnUnitActiveSec=1d
-# Daily at 8am
-OnCalendar=*-*-* 18:30:00
+<% include stdlib -%>
```
Comment (bastelfreak) [inline templates/borg-backup.timer.epp:10]: I don't think you need to include stdlib here?

File: templates/borg-backup.timer.epp [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
+<% $backuptime.keys.each | $key | { -%>
+# Daily named job "<%= $key %>" at <%= $backuptime[$key] %>
+OnCalendar=*-*-* <%= $backuptime[$key] %>
```
Comment (bastelfreak) [inline templates/borg-backup.timer.epp:13]: can you please add a unit test that verifies the file content is correct?

File: templates/borg-backup.timer.epp (hunk 3 above)
Comment (sebastianberm) [reply templates/borg-backup.timer.epp:10]: You're right, this should have been removed, I will update this.

File: templates/borg-backup.timer.epp (hunk 4 above)
Comment (sebastianberm) [reply templates/borg-backup.timer.epp:13]: I have added a couple of unit tests to verify the syntax and processing (also processing of multiple backuptimes), but I am not experienced with rspec-puppet to be able to actually verify the correct OnCalendar output.
Would you be able to help me a bit with this (or direct me to the documentation on how I can verify the File in a dependant function call * )?

* What I mean is, the camptocamp/systemd is a dependancy of this module, which is not a problem, but I don't know how to verify the content of the EPP with this.

**Comment by bastelfreak** [comment]
Hi @sebastianberm, thanks for the PR. I rebased it and replaced the windows line endings with unix ones and resubmitted the code as https://github.com/voxpupuli/puppet-borg/pull/111

**Comment by sebastianberm** [comment]
Hi @bastelfreak ,

Thanks!

I'll send in a new PR with support for SSH jumphosts soon :-).

**Pull request closed without merging**